rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -106,7 +112,16 @@
     @Context
     private UriInfo uriInfo;
 
-    private EntityDetail fromEntity(Entity entity) {
+    private EntitySummary fromEntity(Entity entity, boolean includeTags, int detailDepth, List<String> extraSensorGlobs, List<String> extraConfigGlobs) {
+        if (detailDepth==0) {
+            return new EntitySummary(
+                entity.getId(), 
+                entity.getDisplayName(),
+                entity.getEntityType().getName(),
+                entity.getCatalogItemId(),
+                MutableMap.of("self", EntityTransformer.entityUri(entity, ui.getBaseUriBuilder())) );
```

Comment:
An observation - not something necessary but I'd suggest let's add the `self` link into the entity in all cases; this would be more consistent and would mean you could always just use the link if you wanted it, rather than have to check whether it is there and calculate it if not.  It would just mean an extra parameter in the entity detail constructor at 161 below.

Reply:
good call, agree